testing_guide.md

```diff
@@ -0,0 +1,698 @@
+
+> :flashlight: Want to dive deeper into the changes? Check the [PR Review Club Meeting for PR#19602](https://bitcoincore.reviews/19602)
+
+We are going to test this migration mechanism by using the new `migratewallet` RPC to migrate our legacy wallets. There is an extremely large number of possible configurations for the scripts that Legacy wallets can know about, be watching for, and be able to sign for. This guide cannot cover them all. Therefore **testing configurations other than the testing scenarios of this guide is encouraged and could provide valuable feedback to strengthen this migration mechanism**.
+
+We will test the `migratewallet` RPC by creating a new legacy wallet and then migrating it. It is not required to have funds in it as the migration deals with keys and scripts, not really transactions. Even without funds, you can check that the keys and scripts were migrated correctly.
+
+### What to expect after the migration
+
+A new descriptor wallet creates 8 descriptors by default, 2 (external, internal) for each supported script type:
+
+- P2PKH: `pkh([<fingerprint>/44'/1'/0']<xpub>/{0,1}/*)#<checksum>`
+- P2SH(P2WPKH): `sh(wpkh([<fingerprint>/49'/1'/0']<xpub>/{0,1}/*))#<checksum>`
+- P2WPKH: `wpkh([<fingerprint>/84'/1'/0']<xpub>/{0,1}/*)#<checksum>`
+- P2TR: `tr([<fingerprint>/86'/1'/0']<xpub>/{0,1}/*)#<checksum>`
+
+> :information_source: You can verify that with the `listdescriptors` RPC on a new descriptor wallet. For more information on output descriptors, see [doc/descriptors.md][descriptors.md].
+
+A legacy wallet with an HD chain has an HD seed. This seed is hashed according to the BIP32 specification to become the BIP32 master key which everything else is then derived from. We need to associate those addresses with a descriptor. Because of that, in addition to the above, the new migrant wallet is expected to have:
+
+- A range combo descriptor for the external addresses: `combo(<xpub>/0'/0'/*')#<checksum>`
+- A range combo descriptor for the internal addresses: `combo(<xpub>/0'/1'/*')#<checksum>`
+- A combo descriptor for the hd seed: `combo(<hdseed>)#<checksum>`
+
+> :information_source: The HD seed is a valid key that could receive Bitcoin to it even though it's corresponding addresses would never be given out.
+
+For wallets containing non-HD keys, each key will have its own combo descriptor.
+
+After a successful migration, all the above descriptors will be part of a newly created Descriptor wallet named as the original wallet.
+
+### Preparing a Legacy wallet
+
+Create a legacy test-wallet. This is the wallet we will migrate [later](#migrating-a-legacy-wallet) in the guide.
+
+```shell
+cli -named createwallet wallet_name=legacy descriptors=false
```

Comment:
You don’t have the bdb dependency installed 🚔🚨🚔

Reply:
doh! :woman_facepalming:  thank you! I'll test this later tonight. It will probably lead to a suggestion to add this to the Preparation section, but I will wait till I can confirm my findings. :+1:

Reply:
It works! I just left a [comment](https://github.com/kouloumos/rc24-guide-feedback-repo/pull/1/files#r982986596) in an earlier section (same area that sqlite and qt5 are mentioned) about giving the user a heads up that they will need berkeleydb installed. Feel free to resolve this one!